Scripts/RemediationScripts/Remediate-AddDiagnosticSettingsForLogicApps.ps1

```diff
@@ -0,0 +1,596 @@
+    }
+
+    # Backing up Logic App details.
+    $backupFile = "$($backupFolderPath)\LogicAppsWithoutRequiredDiagnosticSetting.csv"
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 3 of 4] Backing up Logic App details to $($backupFile)"
+
+    $logicAppsWithoutRequiredDiagnosticSetting | Export-CSV -Path $backupFile -NoTypeInformation
+
+    if (-not $DryRun)
+    {
+        Write-Host "*** There will be billing cost associated with adding Diagnostic Setting for Logic Apps. ***" -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host "*** In each resource group having Logic App resources, new Storage Accounts will be created to store diagnostic settings related data. ***" -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host "Do you still want to proceed?" -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+
+        $userInput = Read-Host -Prompt "(Y|N)"
+
+        if ($userInput -ne "Y")
+        {
+            Write-Host "Diagnostic setting will not be added to any Logic App. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+            break
+        }
+
+        Write-Host $([Constants]::DoubleDashLine)
+        Write-Host "[Step 4 of 4] Adding Diagnostic setting for Logic Apps..." -ForegroundColor $([Constants]::MessageType.Warning)
+
+        # To hold results from the remediation.
+        $remediatedLogicApps = @()
+
+        # Includes Logic Apps that were skipped during remediation due to any errors.
+        $skippedLogicApps = @()
+
+        #Write-Host "Checking if Auditing and Advanced Threat Protection are configured for the individual SQL servers." -ForegroundColor $([Constants]::MessageType.Info)
+
+        # Storage Account details
+        [String] $storageAccountResourceGroupName = [String]::Empty
+        [String] $storageAccountName = [String]::Empty
+        $storageAccount = $null
+        #$isStorageAccountPreferenceDecided = $false
+        
+        Write-Host "Adding diagnostic settng requires one or more of Storage Account, Log Analytics Workspace or Event Hub to be configured for storing diagnostic logs." -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host "*** This script supports only Storage Accounts as a destination for storing the diagnostic logs. ***" -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host "Storage Accounts will be created per Resource Group and Location combination to store the diagnostic logs of all Logic Apps in the Resource Group" -ForegroundColor $([Constants]::MessageType.Info) 
+        Write-Host "Do you still want to proceed?" -ForegroundColor $([Constants]::MessageType.Info) -NoNewline
+
+        $userInput = Read-Host -Prompt "(Y|N)"
+
+        if ($userInput -ne "Y")
+        {
+            #Write-Host "If you prefer a different destination for storing the diagnostic logs, please configure them using Azure " -ForegroundColor $([Constants]::MessageType.Update)
```

Comment:
any reason for this comment?

Reply:
removed this comment and added warning as this script supports only storage account as destination for diagnostic setting